core/src/test/kotlin/io/specmatic/core/ScenarioTest.kt

```diff
@@ -658,4 +662,254 @@ class ScenarioTest {
+
+            val paths = scenario.calculatePath(httpRequest)
+
+            assertThat(paths).containsExactly("{TestRequest}.data{string}")
+        }
+
+        @Test
+        fun `calculatePath should handle AnyPattern body with scalar wrapping`() {
+            val scenario = Scenario(
+                name = "test",
+                httpRequestPattern = HttpRequestPattern(
+                    body = AnyPattern(listOf(StringPattern(), NumberPattern()))
+                ),
+                httpResponsePattern = HttpResponsePattern(
+                    headersPattern = HttpHeadersPattern(),
+                    status = 200,
+                    body = StringPattern()
+                )
+            )
+            val httpRequest = HttpRequest(
+                method = "POST",
+                path = "/test",
+                body = StringValue("test")
+            )
+
+            val paths = scenario.calculatePath(httpRequest)
+
+            assertThat(paths).containsExactly("{string}")
+        }
+
+        @Test
+        fun `calculatePath should handle AnyPattern body with non-scalar types`() {
+            val objectPattern = JSONObjectPattern(
+                mapOf("field" to StringPattern()),
+                typeAlias = "(CustomObject)"
+            )
+            val scenario = Scenario(
+                name = "test",
+                httpRequestPattern = HttpRequestPattern(
+                    body = AnyPattern(listOf(objectPattern))
+                ),
+                httpResponsePattern = HttpResponsePattern(
+                    headersPattern = HttpHeadersPattern(),
+                    status = 200,
+                    body = StringPattern()
+                )
+            )
+            val httpRequest = HttpRequest(
+                method = "POST",
+                path = "/test",
+                body = JSONObjectValue(mapOf("field" to StringValue("test")))
+            )
+
+            val paths = scenario.calculatePath(httpRequest)
+
+            assertThat(paths).containsExactly("CustomObject")
+        }
+
+        @Test
+        fun `calculatePath should handle ListPattern body`() {
+            val scenario = Scenario(
+                name = "test",
+                httpRequestPattern = HttpRequestPattern(
+                    body = ListPattern(AnyPattern(listOf(StringPattern(), NumberPattern())))
+                ),
+                httpResponsePattern = HttpResponsePattern(
+                    headersPattern = HttpHeadersPattern(),
+                    status = 200,
+                    body = StringPattern()
+                )
+            )
+            val httpRequest = HttpRequest(
+                method = "POST",
+                path = "/test",
+                body = JSONArrayValue(listOf(StringValue("test"), NumberValue(42)))
+            )
+
+            val paths = scenario.calculatePath(httpRequest)
+
+            assertThat(paths).containsExactlyInAnyOrder("[0]{string}", "[1]{number}")
```

Comment:
This assertion should be:

assertThat(paths).containsExactlyInAnyOrder("{[0]}{string}", "{[1]}{number}")

the 0 and 1 are top-level objects, and since they don't have aliases, their index should appear within square brackets nested within braces, as in the given example.